File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    String.downcase(sentence)
    |> String.split(~r/[^[:alnum:]-]/u, trim: true)
    |> Enum.reduce(%{}, fn elem, acc ->
      Map.update(acc, elem, 1, &(&1 + 1))
    end)
  end
end

File: typescript/word-count/word-count.ts
export function count(sentence: string): Map<string, number> {
  return sentence
    .trim()
    .split(/[\t\s\n]+/)
    .map((word) => word.toLocaleLowerCase())
    .reduce((map, current) => {
      return map.set(current, map.get(current) + 1 || 1);
    }, new Map());
}
